0038-http-auth-profiles/http-auth-profiles.md

```diff
@@ -0,0 +1,183 @@
+---
+title: Interledger HTTP Authentication Profiles
+draft: 1
+---
+
+# Interledger HTTP Authentication Profiles
+A minimal set of extensible, token-based profiles for securing Interledger relationships using HTTP.
+
+## Motivation
+Many Interledger protocols and software work well using HTTP. For example, [ILP-over-HTTP](https://github.com/interledger/rfcs/blob/master/0035-ilp-over-http/0035-ilp-over-http.md) utilizes HTTP at the [Link Layer](https://interledger.org/rfcs/0001-interledger-architecture/). Additionally, Interledger software often uses HTTP for administrative functionality and other purposes.
+
+This document defines common profiles that can be used to ensure implementations of Interledger software work securely in an interoperable fashion, while simultaneously allowing Interledger software operators to more clearly understand the trade-offs involved with various security choices.
+
+The profiles defined in this RFC should:
+ 
+* ...clearly define any trade-offs involved in choosing a particular profile.
+* ...be easily implementable across programming languages in order to aid broad adoption.
+* ...utilise well-known and broadly deployed protocols in order to increase confidence in the profiles.
+* ...be suitable for high-performance use-cases, such as ILPv4 packet processing.
+* ...resist replays (emphasis: resist, not necessarily be replay-proof, in order to support high-performance use-cases).
+* ...not negate or prevent the ability of operators to layer-on more security "on top of" a profile defined in this RFC.
+
+## Specification
+
+### Overview
+This document defines three authentication profiles for securing HTTP network connections in Interledger.
+
+### Authentication
+When authenticating requests between Interledger nodes, it is important to choose an algorithm that maintains an appropriate balance between usability and security, while at the same time maintaining high-performance _and_ interoperability.
+
+In order to find this balance, this document defines three Authentication profiles, each with various trade-offs that should be considered before use:
+
+* `SIMPLE`: Allows two ILP nodes to utilize a previously agreed-upon shared-secret as a [Bearer token](https://tools.ietf.org/html/rfc6750) in all HTTP requests. Peers SHOULD consider this token to be opaque and SHOULD NOT derive any special meaning from the token. 
+
```

Comment:
> Out of interest, the OAuth provider we have used for our POC does not use JWT's as access tokens, it uses an opaque token and has some security reasons why they recommend that but I can't find those docs right now!

@adrianhopebailie Moving this comment to a discussion here so we can resolve it once we're done. 

So, I think both models can work. There's just a tradeoff to be made during the "introspection" step. Either 1) a token can be validated in-memory without calling another system (JWT does this in a widely deployed and understood fashion), or 2) the Resource Server needs to call out to an Auth system or database and ask it if the token is valid. 

[Hydra Oathkeer](https://www.ory.sh/docs/oathkeeper/) is an interesting example here. It's an Auth proxy so it introspects the opaque non-JWT token and makes an Auth decision, but then the upstream service still needs to know how to trust the positive or negative assertion here. I can't tell how it does that, but it seems flexible.